images/win/scripts/Installers/Download-ToolCache.ps1

```diff
@@ -8,18 +8,46 @@ Function Install-NpmPackage {
     [CmdletBinding()]
     param(
         [Parameter(Mandatory=$true)]
-        [System.String]
-        $Name,
+        [System.String] $PackageName,
         [Parameter(Mandatory=$true)]
-        [System.String]
-        $NpmRegistry
+        [System.Uri] $FeedPrefix
     )
 
-    Write-Host "Installing npm '$Name' package from '$NpmRegistry'"
+    Push-Location -Path $env:TEMP
 
-    npm install $Name --registry=$NpmRegistry
+    $FeedUri = $FeedPrefix.AbsoluteUri
+
+    Write-Host "Installing npm $PackageName from ${FeedUri}"
+    npm install $PackageName --registry "${FeedUri}"
+
+    $exitCode = $LASTEXITCODE
+    if($exitCode -ne 0) {
+        Write-Host "$PackageName installation failure;  Error: ${exitCode}"
+
+        exit 1
+    }
+
+    Pop-Location
 }
 
+Function NPMFeed-Auth {
```

Comment:
It's not an actual auth, it is auth setup, right? If so, let's rename the method to NPMFeed-AuthSetup or so.

Reply:
Done.